shared/src/main/scala/coop/rchain/catscontrib/BooleanOps.scala

```diff
@@ -8,7 +10,7 @@ final class BooleanOps(self: Boolean) {
 
   final class ConditionalEither[A](a: => A) {
     def or[B](b: => B): Either[B, A] =
-      if (self) Right(a) else Left(b)
+      if (self) a.asRight[B] else b.asLeft[A]
```

Comment:
What's wrong with `if (self) Right(a) else Left(b)`?

Reply:
https://www.scala-lang.org/api/2.12.6/scala/util/Right.html

Right(a) does not even give you `Either[Nothing, A]`, it actually gives you `Right[Nothing, A]`
while `Left(b)` gives you `Left[B, Nothing]`

Reply:
I see. So it's better type inference (`Either[B, A]` in this case) vs some ops-wrapping overhead which is negligible most of the times. I guess we should strive for the better ergonomics (inference!) and override as needed for performance (if at all) :+1: